src/operations/bulk_write.ts

```diff
@@ -24,18 +25,15 @@ export class BulkWriteOperation extends OperationBase<BulkWriteOptions, BulkWrit
 
     this.collection = collection;
     this.operations = operations;
+
+    // Assign BSON serialize options to OperationBase, preferring options over collection options
+    Object.assign(this, inheritBSONOptions(options, collection.s));
```

Comment:
Total opinons here:

I'd love some consistency if possible. It seems that collection has it on `.s` but bulk_write is on the class itself.

Also using `Object.assign(this...` is super convenient  but usually shy away from it to ensure types not being violated. 

Here I'd prefer:

```ts
this.bsonOptions = inheritBSONOptions(options, collection.s));
```

or 

```ts
const bsonOptions = inheritBSONOptions(options, collection.s));
this.raw = bsonOptions.raw;
this.promoteLongs = bsonOptions.promoteLongs;
this.promoteValues = bsonOptions.promoteValues;
this.promoteBuffers = bsonOptions.promoteBuffers;
this.promoteBuffers = bsonOptions.promoteBuffers;
this.ignoreUndefined = bsonOptions.ignoreUndefined;
this.serializeFunctions = bsonOptions.serializeFunctions;
this.fieldsAsRaw = bsonOptions.fieldsAsRaw;
```

Reply:
Your point about consistency is great, I definitely struggled with that for these changes. 

It seems to me that for the Collection/Db/Client, the bson options must be somewhere in `.s`, but classes extending `OperationBase` don't (currently) have a `.s`, so unless that changes there can't be perfect consistency...

Here is how I was thinking about it (assuming I make the change for `Collection`): `Collection`, `Db`, and `MongoClient` could all store the bson options in `s.options`, while anything extending `OperationBase` has the bson options on the class itself. The rational for the second part of this seemed to come out of @nbbeeken's work-- wondering if he has any thoughts on this.